common/v2/features/SendAssets/components/fields/EthAddressField.tsx

```diff
@@ -54,26 +57,44 @@ function ETHAddressField({
         name={fieldName}
         validate={validateEthAddress}
         render={({ field, form }: FieldProps) => (
-          <Input
-            {...field}
-            placeholder={placeholder}
-            onBlur={e => {
-              if (chainId) {
-                const ensTLD = getENSTLDForChain(chainId);
-                const isENSAddress = e.currentTarget.value.endsWith(`.${ensTLD}`);
-                form.setFieldValue('resolvedNSAddress', '');
-                if (isENSAddress && handleENSResolve) {
-                  handleENSResolve(e.currentTarget.value);
-                  handleGasEstimate();
-                } else {
-                  form.setFieldValue('receiverAddress', e.currentTarget.value);
-                  handleGasEstimate();
+          <div>
+            <Input
+              {...field}
+              value={field.value.display}
+              placeholder={placeholder}
+              onChange={e => {
+                form.setFieldValue('receiverAddress', {
+                  display: e.currentTarget.value,
+                  value: e.currentTarget.value
+                });
+              }}
+              onBlur={e => {
+                if (network.chainId) {
+                  const ensTLD = getENSTLDForChain(network.chainId);
+                  const isENSAddress = e.currentTarget.value.endsWith(`.${ensTLD}`);
+                  if (isENSAddress && handleENSResolve) {
+                    handleENSResolve(e.currentTarget.value);
+                    handleGasEstimate();
+                  } else {
+                    form.setFieldValue('receiverAddress', {
+                      display: e.currentTarget.value,
+                      value: e.currentTarget.value
+                    });
+                    handleGasEstimate();
+                  }
```

Comment:
No nested conditionals
```
e => {
  if (!network.chainId) return;
  const ensTLD = getENSTLDForChain(network.chainId);
  const isENSAddress = e.currentTarget.value.endsWith(`.${ensTLD}`);
  if (isENSAddress && handleENSResolve) {
    handleENSResolve(e.currentTarget.value);
  } else {
    form.setFieldValue('receiverAddress', {
       display: e.currentTarget.value,
       value: e.currentTarget.value
   });
  }
  handleGasEstimate(); // this is called in any case so you can extract it from the conditional block
}

```

And since both `handleENSResolve` and `setFieldValue` are async and I suspect you want `handleGasEstimate` to be called with the new values:
e.g https://github.com/jaredpalmer/formik/issues/529#issuecomment-505887509
```
async (e) => {
  if (!network.chainId) return;
  const ensTLD = getENSTLDForChain(network.chainId);
  const isENSAddress = e.currentTarget.value.endsWith(`.${ensTLD}`);
  const action = (isENSAddress && handleENSResolve)
    ? (address) => handleENSResolve(receiverAddress.display)
     : (address) => form.setFieldValue('receiverAddreess', { display: address, value: address })
  await action(e.currentTarget.value);
  handleGasEstimate();
}
```